75 navigation redesign bars

nemislimus opened the pull request:
Переделал RootContainer для отображения topBar и bottomNav и возможностью прятать эти компоненты на нужных экранах.
Для bottomNav сделал вложенные графы. 
Всё проверил. Местами для отображения есть моки.

nemislimus committed:
#75 add deps, res, models

nemislimus committed:
#75 create root container, bottom nav, update models

nemislimus committed:
#75 set nested graphs for BottomNav

nemislimus committed:
#75 done

nemislimus committed:
#75 detekt-fix-1

ShevaVarya commented on app/src/main/java/cy/volleybolley/core/presentation/MainActivity.kt:
```diff
@@ -4,26 +4,47 @@ import android.os.Bundle
-import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.BottomNavigation
```
<img width="608" height="86" alt="image" src="https://github.com/user-attachments/assets/c03f03c2-7107-42ea-aefa-62e52935ca9e" />

ShevaVarya replied:
Почему не material3?

ShevaVarya commented on app/src/main/java/cy/volleybolley/core/presentation/MainActivity.kt:
```diff
@@ -128,17 +86,117 @@ fun RootContainer(
+        TopLevelRoute(
+            stringResource(R.string.my_games),
+            MyGamesTopLevelRoute,
+            painterResource(R.drawable.ic_players),
```
А почему ты мячик players назвад? :D

ShevaVarya commented on app/src/main/java/cy/volleybolley/core/presentation/MainActivity.kt:
```diff
@@ -32,94 +53,31 @@ class MainActivity : ComponentActivity() {
+    val showTopBar = NoBarsRoutes.showTopBar(currentDestinationRoute)
+
+    navController.addOnDestinationChangedListener { _, destination, _ ->
+        currentDestinationRoute = destination.route.toString()
```
Слушай, а тут не может быть null и тогда строка запишется "null", это ок?

Secur1ty-21 replied:
При новом подходе через сериализуемые объекты, не должно быть значения null, кроме старта приложения на короткое мгновение. На всякий добавил логику на случай null скрывать topBar и bottomBar

ShevaVarya commented on app/src/main/java/cy/volleybolley/core/presentation/MainActivity.kt:
```diff
@@ -128,17 +86,117 @@ fun RootContainer(
+            val itemIsSelected = currentDestination?.hierarchy?.any { it.hasRoute(topRoute.route::class) } == true
+            val labelStyle = if (itemIsSelected) BodyTinyBottomNavGradient else BodyTinyBottomNavWhite
+
+            BottomNavigationItem(
```
У тебя отступ от элемента 8dp, а по дизайну 10

ShevaVarya commented on app/src/main/java/cy/volleybolley/core/presentation/ui/component/VolleyTopBar.kt:
```diff
@@ -59,27 +60,27 @@ object VolleyTopBar {
-            shadowElevation = 4.dp
+            shadowElevation = VolleyDimens.DIMEN_4.dp
         ) {
             Box(modifier = Modifier.fillMaxSize()) {
```
Вроде избыточный fillMaxSize

ShevaVarya reviewed (approved):
Потыкала, все работает, ты молодец! Небольшое замечание по верстке, и может в превью сделать MainActivity, а не LaunchScreen?

Commit:
#80 Исправление замечаний по ревью
1. Замена BottomBar с material на material3
2. Исправлен отступ

Secur1ty-21 reviewed (approved):
Отличная работа, замечаний нет

Commit:
#80 Исправление замечаний по ревью
1. Добавление логики скрытия bottomBar и topBar в случае получения destination?.route = null

Commit:
#80 Исправление замечаний по ревью
1. Добавление логики скрытия bottomBar и topBar в случае получения destination?.route = null

Commit:
#75 Чистка дублирования вызова currentBackStackEntryAsState, в двух разный функциях
